linc_convert/modalities/lsm/transfer.py

```diff
@@ -0,0 +1,106 @@
+"""
+Convert a collection of spool .dat files generated by light sheet microscopy to compressed tar files and upload with the DANDI client.
+"""
+
+# stdlib
+import os
+from datetime import datetime
+from pathlib import Path
+
+# externals
+import tarfile
+import cyclopts
+import dandi.download
+import dandi.upload
+
+# internals
+from linc_convert.modalities.lsm.cli import lsm
+
+transfer = cyclopts.App(name="transfer", help_format="markdown")
+lsm.command(transfer)
+
+@transfer.default
+def dandi_transfer(input_dir, dandiset_url, dandi_instance, subject, output_dir='.',  max_size_gb=2.00, upload=False):
+    """
+    Upload .dat files to DANDI in batched, compressed tar archives.
+    
+    Parameters
+    ----------
+    input_dir : str
+        Directory containing .dat files to upload
```

Comment:
An alternative could be for the input to be a list of files, rather than doing the glob ourselves in the function. Dat files can still be filtered using a command-line glob: `linc-convert lsm transfer path/to/dir/*.dat`. And it would allows taring non-dat files if needed.

But I understand that the directory-based interface might be easier to use for Emin et al.